control-plane/gateways/deployment_init_container.go

```diff
@@ -143,17 +146,25 @@ func initContainer(config GatewayConfig, name, namespace string) (corev1.Contain
 			Value: consulNamespace,
 		})
 
-	if config.ConsulTenancyConfig.ConsulPartition != "" {
+	if b.config.ConsulTenancyConfig.ConsulPartition != "" {
 		container.Env = append(container.Env,
 			corev1.EnvVar{
 				Name:  "CONSUL_PARTITION",
-				Value: config.ConsulTenancyConfig.ConsulPartition,
+				Value: b.config.ConsulTenancyConfig.ConsulPartition,
 			})
 	}
 
 	return container, nil
 }
 
+func initContainerResourcesOrDefault(gcc *meshv2beta1.GatewayClassConfig) corev1.ResourceRequirements {
+	if gcc != nil && gcc.Spec.Deployment.InitContainer != nil && gcc.Spec.Deployment.InitContainer.Resources != nil {
+		return *gcc.Spec.Deployment.InitContainer.Resources
+	}
+
+	return corev1.ResourceRequirements{}
```

Comment:
I'm assuming if this is empty k8s just uses whatever the defaults are?

Reply:
Yeah, it just sets it to empty and there are no requests/limits. It's not recommended, but we can set real defaults later.